pkg/backup/backupinfo/desc_sst.go

```diff
@@ -0,0 +1,247 @@
+// Copyright 2025 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package backupinfo
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"sort"
+
+	"github.com/cockroachdb/cockroach/pkg/backup/backuppb"
+	"github.com/cockroachdb/cockroach/pkg/cloud"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/sql/catalog/descpb"
+	"github.com/cockroachdb/cockroach/pkg/storage"
+	"github.com/cockroachdb/cockroach/pkg/util/encoding"
+	"github.com/cockroachdb/cockroach/pkg/util/protoutil"
+	"github.com/cockroachdb/errors"
+)
+
+const (
+	// BackupMetadataDescriptorsListPath is the name of the SST file containing
+	// the BackupManifest_Descriptors or BackupManifest_DescriptorRevisions of the
+	// backup. This file is always written in conjunction with the
+	// `BACKUP_METADATA`.
+	BackupMetadataDescriptorsListPath = "descriptorslist.sst"
+	sstDescsPrefix                    = "desc/"
+)
+
+// WriteDescsSST is responsible for writing the SST containing the Descriptor
+// and DescriptorChanges field of the input BackupManifest. If DescriptorChanges
+// is non-empty, then the descriptor changes will be written to the SST with the
+// MVCC timestamp equal to the revision time. Otherwise, contents of the
+// Descriptors field will be written to the SST with an empty MVCC timestamp.
+func WriteDescsSST(
```

Comment:
one slightly sad part of this refactor is we lost a (fairly trivial) unit test that directly exercises these iterators. I'm fine merging the refactor for now, but we should follow up with a unit test for these iterators.

Reply:
There are iterator tests in TestManifestHandlingIteratorOperations. They look good enough that I don't think we need a new set of tests.

Reply:
oh nice!